compiler/src/scala/quoted/runtime/impl/QuoteMatcher.scala

```diff
@@ -206,12 +206,35 @@ object QuoteMatcher {
         // Matches an open term and wraps it into a lambda that provides the free variables
         case Apply(TypeApply(Ident(_), List(TypeTree())), SeqLiteral(args, _) :: Nil)
             if pattern.symbol.eq(defn.QuotedRuntimePatterns_higherOrderHole) =>
+
+          /* Some of method symbols in arguments of higher-order term hole are eta-expanded.
+           * e.g.
+           * g: (Int) => Int
+           * => {
+           *   def $anonfun(y: Int): Int = g(y)
+           *   closure($anonfun)
+           * }
+           *
+           * f: (using Int) => Int
+           * => f(using x)
+           * This function restores the symbol of the original method from
+           * the eta-expanded function.
+           */
+          def getCapturedIdent(arg: Tree)(using Context): Ident =
+            arg match
+              case id: Ident => id
+              case Apply(term, _) => getCapturedIdent(term)
+              case Block((ddef: DefDef) :: _, _) => getCapturedIdent(ddef.rhs)
```

Comment:
```suggestion
              case Block((ddef: DefDef) :: _, _: Closure) => getCapturedIdent(ddef.rhs)
```

Reply:
Thank you, added in b120ffb32d40d6b684fdceacf3ff890fe984fad6